From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Fabi.exe" <66797305+FabiPunktExe@users.noreply.github.com>
Date: Mon, 24 Jun 2024 16:18:38 +0200
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 8c2c5ea14a839c8767bbf120ae5512f65475cbb0..5681f94d6909bbd090376c084ee802a11bbb7c2c 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -32,6 +32,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // Grindstone start
+    /**
+     * The brand id for Grindstone.
+     */
+    Key BRAND_GRINDSTONE_ID = Key.key("diruptio", "grindstone");
+    // Grindstone end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index c880d0010849ab733ad13bbd18fab3c864d0cf61..dbc235ad8f6a255509c81319e0c74aa3c3f74e46 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -259,7 +259,7 @@ public class VersionCommand extends BukkitCommand {
         // Purpur start
         int distance = getVersionFetcher().distance();
         final Component message = Component.join(net.kyori.adventure.text.JoinConfiguration.separator(Component.newline()),
-            ChatColor.parseMM("<grey>Current Purpur Version: %s%s*", distance == 0 ? "<green>" : distance > 0 ? "<yellow>" : "<red>", Bukkit.getVersion()),
+            ChatColor.parseMM("<grey>Current Grindstone Version: %s%s*", distance == 0 ? "<green>" : distance > 0 ? "<yellow>" : "<red>", Bukkit.getVersion()), // Grindstone
             // Purpur end
             msg
         );
