From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Fabi.exe" <66797305+FabiPunktExe@users.noreply.github.com>
Date: Sun, 30 Jun 2024 10:25:03 +0200
Subject: [PATCH] Disable item merging


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 1d594c256200d46882fafce5d2b33e60f0fa3a59..72bedd18cef22ac903ee5164e30948a68accbf00 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2266,6 +2266,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Grindstone start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getGrindstoneConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Grindstone end
+
         /**
          * Sends the component to the player
          *
