From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Fabi.exe" <66797305+FabiPunktExe@users.noreply.github.com>
Date: Sun, 30 Jun 2024 10:25:03 +0200
Subject: [PATCH] Disable item merging


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 84b07a7fd7941b2f75b531646b67a124a0607390..d9729ad0cdbc7f56715edc1589ae3df30fd6e0b0 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2277,6 +2277,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Grindstone start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getGrindstoneConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Grindstone end
+
         /**
          * Sends the component to the player
          *
