From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Fabi.exe" <66797305+FabiPunktExe@users.noreply.github.com>
Date: Sun, 30 Jun 2024 10:32:13 +0200
Subject: [PATCH] Disable item base tick


diff --git a/src/main/java/diruptio/grindstone/GrindstoneConfig.java b/src/main/java/diruptio/grindstone/GrindstoneConfig.java
index f2661e5cb9573479e5116b4eab0f10f15c608091..747d729c3c70045298c016c27b9e447d231367dc 100644
--- a/src/main/java/diruptio/grindstone/GrindstoneConfig.java
+++ b/src/main/java/diruptio/grindstone/GrindstoneConfig.java
@@ -128,10 +128,12 @@ public class GrindstoneConfig {
     }
 
     public static boolean disableItemMerging = false;
+    public static boolean disableItemBaseTick = false;
 
     private static void entityOptions() {
         if (version >= 1) {
             disableItemMerging = getBoolean("entity.item.disable-merging", disableItemMerging);
+            disableItemBaseTick = getBoolean("entity.item.disable-base-tick", disableItemBaseTick);
         }
     }
 }
diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index 0f964bd9dfc90cd2327efe85f5c1b06a08d46297..51d30abffaa7eb8bf75b0b60c155f843ff7f5983 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -158,7 +158,9 @@ public class ItemEntity extends Entity implements TraceableEntity {
         if (this.getItem().isEmpty()) {
             this.discard(EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
         } else {
+            if (!diruptio.grindstone.GrindstoneConfig.disableItemBaseTick) { // Grindstone
             super.tick();
+            } // Grindstone
             // Paper start - remove anti tick skipping measures / wall time - revert to vanilla
             if (this.pickupDelay > 0 && this.pickupDelay != 32767) {
                 --this.pickupDelay;
