From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Fabi.exe" <66797305+FabiPunktExe@users.noreply.github.com>
Date: Sun, 30 Jun 2024 10:32:13 +0200
Subject: [PATCH] Disable item base tick


diff --git a/src/main/java/diruptio/grindstone/GrindstoneConfig.java b/src/main/java/diruptio/grindstone/GrindstoneConfig.java
index f0187f24c8892a994e4a7061f7f70b798641100b..6b92202a51e9f1aba5e6bd14f0c8990ff64acdba 100644
--- a/src/main/java/diruptio/grindstone/GrindstoneConfig.java
+++ b/src/main/java/diruptio/grindstone/GrindstoneConfig.java
@@ -128,10 +128,12 @@ public class GrindstoneConfig {
     }
 
     public static boolean disableItemMerging = false;
+    public static boolean disableItemBaseTick = false;
 
     private static void entityOptions() {
         if (version >= 1) {
             disableItemMerging = getBoolean("entity.item.disable-merging", disableItemMerging);
+            disableItemBaseTick = getBoolean("entity.item.disable-base-tick", disableItemBaseTick);
         }
     }
 }
diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
index e3412f02a228affd2918678a61bcf43451b6d4f0..841317149d850166dd83e21d554869908f5c911c 100644
--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
@@ -158,7 +158,9 @@ public class ItemEntity extends Entity implements TraceableEntity {
         if (this.getItem().isEmpty()) {
             this.discard(EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
         } else {
+            if (!diruptio.grindstone.GrindstoneConfig.disableItemBaseTick) { // Grindstone
             super.tick();
+            } // Grindstone
             // CraftBukkit start - Use wall time for pickup and despawn timers
             int elapsedTicks = MinecraftServer.currentTick - this.lastTick;
             if (this.pickupDelay != 32767) this.pickupDelay -= elapsedTicks;
